lib/auth/accesslist/service.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslist
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	conv "github.com/gravitational/teleport/api/convert/teleport/accesslist/v1"
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+)
+
+// ServiceConfig is the service config for the Okta gRPC service.
+type ServiceConfig struct {
+	// Backend is the backend to use.
+	Backend backend.Backend
+
+	// Logger is the logger to use.
+	Logger logrus.FieldLogger
+
+	// Authorizer is the authorizer to use.
+	Authorizer authz.Authorizer
+
+	// AccessLists is the access list service to use.
+	AccessLists services.AccessLists
+}
+
+func (c *ServiceConfig) CheckAndSetDefaults() error {
+	if c.Backend == nil {
+		return trace.BadParameter("backend is missing")
+	}
+
+	if c.Logger == nil {
+		c.Logger = logrus.New().WithField(trace.Component, "okta_crud_service")
+	}
+
+	if c.Authorizer == nil {
+		return trace.BadParameter("authorizer is missing")
+	}
+
+	var err error
+	if c.AccessLists == nil {
+		c.AccessLists, err = local.NewAccessListService(c.Backend, c.Backend.Clock())
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+var _ accesslistv1.AccessListServiceServer = (*Service)(nil)
+
+type Service struct {
```

Comment:
Suggestion: In the future I'd push this to a separate PR from the protos, so design comments aren't mixed with impl.